### PR TITLE
UI: popup menu updates

### PR DESCRIPTION
**Pull request by hellobontempo**
Screenshot 2024-02-07 at 8 52 49 AM

## enterprise tests ✅

### CODE REVIEW

File: ui/app/templates/vault/cluster/policies/index.hbs [hunk 1]
```diff
@@ -91,52 +91,39 @@
...
-                          @confirmTitle="Delete this policy?"
-                          @confirmMessage="This will permanently delete this policy and may affect access to some data"
-                          @onConfirmAction={{action "deletePolicy" item}}
-                        />
-                      {{/if}}
-                    {{/if}}
-                  </ul>
-                </nav>
-              </PopupMenu>
+              <Hds::Dropdown @isInline={{true}} as |dd|>
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/templates/vault/cluster/policies/index.hbs:94]: Good question! That's the default! So I left it out to shorten the args, but I can add it back for clarity

File: ui/app/templates/vault/cluster/access/namespaces/index.hbs [hunk 2]
```diff
@@ -35,32 +35,51 @@
...
+        {{#if this.namespaceToDelete}}
+          <ConfirmModal
+            @color="critical"
             @confirmTitle="Delete this namespace?"
             @confirmMessage="Any engines or mounts in this namespace will also be removed."
-            @onConfirmAction={{action
+            @onClose={{fn (mut this.namespaceToDelete) null}}
+            @onConfirm={{action
               (perform
                 Item.callMethod
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/templates/vault/cluster/access/namespaces/index.hbs:73]: I know 😭

File: ui/app/components/identity/popup-alias.js
Comment (Monkeychip) [inline ui/app/components/identity/popup-alias.js:1]: way to throw in some glimmer work ✨

Comment (hellobontempo) [reply ui/app/components/identity/popup-alias.js:1]: @hashishaw rocked that! 🥇

File: ui/app/components/identity/popup-members.js [hunk 3]
```diff
@@ -3,37 +3,44 @@
...
+    if (this.args.onError) {
+      this.args.onError();
+    }
+    const error = errorMessage(err);
+    this.flashMessages.danger(`There was a problem removing '${memberId}' from the group - ${error}`);
+  }
+
+  transaction() {
+    const members = this.args.model[this.args.groupArray];
+    this.args.model[this.args.groupArray] = members.without(this.args.memberId);
```
Comment (Monkeychip) [inline ui/app/components/identity/popup-members.js:32]: This is more for @chelshaw as Claire and I have already chatted about this. But I figured this might be a decent example of where I'm confused about what we _should not_ be allowed to do in a glimmer component and what we _can do_. Technically `this.args` are `readonly` and we should hit an error when we try and reassign them? —[article](https://v5.chriskrycho.com/journal/understanding-args-in-glimmer-components/) from Chris Krycho who talks about it in detail. Any idea how we're able to do this, I know we do it elsewhere in the app?

File: ui/app/components/identity/popup-policy.js [hunk 4]
```diff
@@ -3,32 +3,47 @@
...
+      this.args.onError();
+    }
+    const error = errorMessage(err);
+    this.flashMessages.danger(`There was a problem removing '${policyName}' policy - ${error}`);
+  }
 
-  successMessage(model, policyName) {
-    return `Successfully removed '${policyName}' policy from ${model.id} `;
-  },
+  transaction() {
```
Comment (Monkeychip) [inline ui/app/components/identity/popup-policy.js:30]: should this be async as you call an await on it below?

File: ui/app/components/identity/popup-members.js (hunk 3 above)
Comment (zofskeez) [reply ui/app/components/identity/popup-members.js:32]: It would throw an error if an arg itself was being reassigned, for example `this.args.model = null`. In this case a property on the arg is mutated. Maybe not best practice but acceptable from a framework standpoint.

Comment (hashishaw) [reply ui/app/components/identity/popup-members.js:32]: Jordan is right -- here we're not updating an arg, we're updating an attribute on an arg. Small but important difference!

File: ui/app/components/identity/popup-policy.js (hunk 4 above)
Comment (hashishaw) [reply ui/app/components/identity/popup-policy.js:30]: Since it's returning a promise (`model.save`) and we're not using `await` within this method there's no need for `async` here

File: ui/app/templates/components/mfa/method-list-item.hbs [hunk 5]
```diff
@@ -30,30 +30,26 @@
...
-                  Edit
-                </LinkTo>
-              </li>
-            </ul>
-          </nav>
-        </PopupMenu>
+        <Hds::Dropdown @isInline={{true}} @listPosition="bottom-right" as |dd|>
+          <dd.ToggleIcon
+            @icon="more-horizontal"
+            @text="Manage M-F-A method"
```
Comment (Monkeychip) [inline ui/app/templates/components/mfa/method-list-item.hbs:36]: Nice catch on accessibility.

File: ui/lib/core/addon/components/confirmation-modal.js
Comment (zofskeez) [inline ui/lib/core/addon/components/confirmation-modal.js:1]: Is this different than the `confirm-modal` then? Curious why we need both if so?

File: ui/app/templates/vault/cluster/access/identity/index.hbs [hunk 6]
```diff
@@ -32,63 +32,53 @@
...
+              @route="vault.cluster.access.identity.show"
+              @models={{array item.id "details"}}
+            />
+            {{#if (or item.isReloading item.updatePath.isPending item.aliasPath.isPending)}}
+              <dd.Generic class="has-text-center">
+                <LoadingDropdownOption />
+              </dd.Generic>
+            {{else}}
+              {{#if item.canAddAlias}}
+                {{! entities can always add aliases, internal groups cannot have any and external groups can only have one }}
```
[the remark this answers is not in the document]
Comment (Monkeychip) [reply ui/app/templates/vault/cluster/access/identity/index.hbs:54]: +1

**Review by zofskeez — APPROVED** [review]

File: ui/app/components/identity/_popup-base.js
[the remark this answers is not in the document]
Comment (hashishaw) [reply ui/app/components/identity/_popup-base.js:1]: I found the logic hard to follow with how this component was extended, and since we were passing spread args around it was also hard to debug what was happening. The extended components all have different arguments so it didn't make sense to extend it anymore. I did consider a decorator, which we could still do as a follow-on, but might be too specific for this small use case

File: ui/app/templates/vault/cluster/policies/index.hbs (hunk 1 above)
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/templates/vault/cluster/policies/index.hbs]: I added it since `@isInline` could inherit styling from the parent, so decided to explicitly define while we're still getting accustom to using the `Hds::Dropdown`

File: ui/lib/core/addon/components/confirmation-modal.js
Comment (hellobontempo) [reply ui/lib/core/addon/components/confirmation-modal.js:1]: `Confirmation-modal` is the "type to confirm" modal. They're similar, but have slightly different functionalities. I think Chelsea opted to make a new `confirm-modal` component to iterate towards cleaning up `ConfirmAction` without blowing out the scope of this PR. The reason being the modal originally rendered inside `ConfirmAction` had to be pulled out because `Hds::Dropdown` is contextual and so we could no longer yield `ConfirmAction` and rely on it to render the modal 

One option is to combine the components and pass a `@typeToConfirm` arg or something.

File: ui/app/components/secret-list/aws-role-item.js
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/components/secret-list/aws-role-item.js:1]: Agreed! I guess we could make a `secret-list` component that has the tracked arg and all of these components extend from it?

File: ui/app/templates/vault/cluster/access/namespaces/index.hbs (hunk 2 above)
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/templates/vault/cluster/access/namespaces/index.hbs:73]: Moved the other occurrences of this outside ListView and defined the task in the relative controller file because tests were mysteriously failing. Left this one because `refreshNamespaceList` wasn't working properly and this PR already has too much going on and just needs to end 😭
